apps/designer/app/designer/shared/design-tokens-manager/token-editor.tsx

```diff
@@ -0,0 +1,179 @@
+import { useState, useEffect, type FormEvent, useRef } from "react";
+import {
+  Button,
+  Flex,
+  InputErrorsTooltip,
+  Label,
+  Popover,
+  PopoverContent,
+  PopoverHeader,
+  PopoverPortal,
+  PopoverTrigger,
+  TextArea,
+  TextField,
+} from "@webstudio-is/design-system";
+import { PlusIcon } from "@webstudio-is/icons";
+import type { DesignToken } from "@webstudio-is/design-tokens";
+import { useDesignTokens } from "~/shared/nano-states";
+import { findByName } from "./utils";
+
+const validate = (
+  tokens: Array<DesignToken>,
+  data: Partial<DesignToken>,
+  isNew: boolean
+): { name: Array<string>; value: Array<string>; hasErrors: boolean } => {
+  const name = [];
+  const value = [];
+
+  if (String(data.name).trim() === "") name.push("Name is required");
+  if (isNew && findByName(tokens, data?.name)) {
+    name.push("Name is already taken");
+  }
+  if (String(data.value).trim() === "") value.push("Value is required");
+
+  return {
+    name,
+    value,
+    hasErrors: name.length !== 0 || value.length !== 0,
+  };
+};
```

Comment:
we use the DesignToken Schema for the db, but I agree it's not 100% the same, it doesn't do trim when checking the length and it doesn't check if name is unique, I will add a todo for this, because we are haven't finalized the design yet, it might all change

Reply:
rewrote the entire validation